packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2104,3 +2139,165 @@ class InputDecoration {
     return 'InputDecoration(${description.join(', ')})';
   }
 }
+
+/// Defines the default appearance of [InputDecorator]s.
+///
+/// This class is used to define the value of [ThemeData.inputDecorationTheme].
+/// The [InputDecorator], [TextField], and [TextFormField] widgets use
+/// the current input decoration theme to initialize null [InputDecoration]
+/// properties.
+@immutable
+class InputDecorationTheme {
+  /// Creates a value for [ThemeData.inputDecorationTheme] that
+  /// defines default values for [InputDecorator].
+  const InputDecorationTheme({
+    this.labelStyle,
+    this.helperStyle,
+    this.hintStyle,
+    this.errorStyle,
+    this.isDense,
+    this.contentPadding,
+    this.isCollapsed,
+    this.prefixStyle,
+    this.suffixStyle,
+    this.counterStyle,
+    this.filled,
+    this.fillColor,
+    this.border : const UnderlineInputBorder(),
+  });
```

Comment:
some of these values must not be null, e.g. isCollapsed, filled; assert and document here